### PR TITLE
Add a parser for California (CAISO)

### PR DESCRIPTION
**Pull request by 7hibault**
Cf #787

### CODE REVIEW

File: parsers/US_CA.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
+# The arrow library is used to handle datetimes
+import arrow
+# The request library is used to fetch content through HTTP
+import requests
+
+from datetime import datetime, timedelta
+
+
+def fetch_production(country_code='FR', session=None):
```
Comment (corradio) [inline parsers/US_CA.py]: should be `US-CA` instead of `FR`

File: parsers/US_CA.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    dailyData = []
+
+    for i in range(0, 24):
+        data = {
+            'countryCode': country_code,
+            'production': {},
+            'storage': {},
+            'source': 'content.caiso.com',
+        }
+        data['countryCode'] = 'US_CA'
```
Comment (corradio) [inline parsers/US_CA.py]: should be `US-CA` instead of `US_CA`.

File: parsers/US_CA.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    obj = response.text.splitlines()
+
+    dailyData = []
+
+    for i in range(0, 24):
+        data = {
+            'countryCode': country_code,
+            'production': {},
+            'storage': {},
+            'source': 'content.caiso.com',
```
Comment (systemcatch) [inline parsers/US_CA.py]: source can just be 'caiso.com' as content.caiso.com redirects there anyway.

File: parsers/US_CA.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+        data['production']['biomass'] = int(obj[i+2].split('\t')[5])
+        data['production']['gas'] = int(obj[i+2].split('\t')[7])
+        data['production']['hydro'] = int(obj[i+2].split('\t')[9]) + int(obj[i+30].split('\t')[11]) # hydro + small hydro
+        data['production']['nuclear'] = int(obj[i+30].split('\t')[5])
+        data['production']['solar'] = int(obj[i+2].split('\t')[13]) + int(obj[i+2].split('\t')[15]) # solar PV + thermal
+        data['production']['wind'] = int(obj[i+2].split('\t')[11])
+        data['production']['geothermal'] = int(obj[i+2].split('\t')[3])
+        data['production']['unknown'] = int(obj[i+2].split('\t')[5])
+        # set the date at the end of the hour
+        dataTime = yesterday.replace(hour = i, minute = 59, second = 0)
```
Comment (systemcatch) [inline parsers/US_CA.py]: Not sure why minute = 59 is here, it can just be on the hour as per the data source.

File: parsers/US_CA.py (hunk 2 above)
Comment (systemcatch) [reply parsers/US_CA.py]: since it's set on L51 anyway it can be removed

File: parsers/US_CA.py [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+
+    for i in range(0, 24):
+        data = {
+            'countryCode': country_code,
+            'production': {},
+            'storage': {},
+            'source': 'content.caiso.com',
+        }
+        data['countryCode'] = 'US_CA'
+        data['datetime'] = datetime.now()
```
Comment (systemcatch) [inline parsers/US_CA.py]: this is set on L69 so setting it here isn't needed

File: parsers/US_CA.py [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+        data['production'] = {}
+        data['production']['biomass'] = int(obj[i+2].split('\t')[5])
```
Comment (systemcatch) [inline parsers/US_CA.py]: if we stick with getting the data like this it'll need to be float rather than int

**Comment by 7hibault** [comment]
Thanks for the review! I didn't know panda's read_table, it's so much cleaner and easier to read... I've tried taking all of your comments into account.

File: config/zones.json [hunk 7]
```diff
@@ -1345,6 +1345,17 @@
       "https://github.com/corradio"
     ]
   },
+  "US-CA": {
```
Comment (corradio) [inline config/zones.json]: US-CA should be after US-AK. Can you order alphabetically?

**Pull request merged**